lib/bigtable/row.js

```diff
@@ -0,0 +1,745 @@
+};
+
+/**
+ * Delete specified cells from the row. See {module:bigtable/table#mutate}.
+ *
+ * @param {string[]} columns - Column names for the cells to be deleted.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * //-
+ * // Delete individual cells.
+ * //-
+ * var callback = function(err, apiResponse) {
+ *   if (!err) {
+ *     // Cells were successfully deleted.
+ *   }
+ * };
+ *
+ * row.deleteCells([
+ *   'follows:gwashington'
+ * ], callback);
+ *
+ * //-
+ * // Delete all cells within a family.
+ * //-
+ * row.deleteCells([
+ *   'follows',
+ * ], callback)
+ */
+Row.prototype.deleteCells = function(columns, callback) {
+  var mutation = {
+    key: this.id,
+    data: arrify(columns),
+    method: Mutation.methods.DELETE
+  };
+
+  this.parent.mutate(mutation, callback);
+};
+
+/**
+ * Get the row data. See {module:bigtable/table#getRows}.
+ *
+ * @param {string[]=} columns - List of specific columns to retrieve.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {module:bigtable/row} callback.row - The updated Row object.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * //-
+ * // Use this method to grab an entire row
+ * //-
+ * var callback = function(err, row, apiResponse) {
+ *   if (!err) {
+ *     // `row.cells` has been updated.
+ *   }
+ * };
+ *
+ * row.get(callback);
+ *
+ * //-
+ * // Or pass in an array of column names to populate specific cells.
+ * // Under the hood this will create an interleave filter.
+ * //-
+ * row.get([
+ *   'follows:gwashington',
+ *   'follows:alincoln'
+ * ], callback);
+ */
+Row.prototype.get = function(keys, callback) {
+  var self = this;
+
+  if (is.function(keys)) {
+    callback = keys;
+    keys = [];
+  }
+
+  var filter;
+
+  keys = arrify(keys);
+
+  if (keys.length) {
+    var filters = keys
+      .map(Mutation.parseColumnName)
+      .map(function(column) {
+        var filters = [{ family: column.family }];
+
+        if (column.qualifier) {
+          filters.push({ column: column.qualifier });
+        }
+
+        return filters;
+      });
+
+    if (filters.length > 1) {
+      filter = [{
+        interleave: filters
+      }];
+    } else {
+      filter = filters[0];
+    }
+  }
+
+  var reqOpts = {
+    key: this.id,
+    filter: filter
+  };
+
+  this.parent.getRows(reqOpts, function(err, rows, apiResponse) {
+    if (err) {
+      callback(err, null, apiResponse);
+      return;
+    }
+
+    var row = rows[0];
+
+    if (!row) {
+      err = new RowError(self.id);
+      callback(err, null, apiResponse);
+      return;
+    }
+
+    extend(true, self.data, row.data);
+    callback(null, keys.length ? row.data : self, apiResponse);
```

Comment:
I'm not sure I understand what's happening here. Can you leave a comment in the code about why if columns were asked for, you're just returning `row.data`, otherwise `self`?